crates/typst-cli/src/world.rs

```diff
@@ -329,8 +347,37 @@ fn system_path(project_root: &Path, id: FileId) -> FileResult<PathBuf> {
     id.vpath().resolve(root).ok_or(FileError::AccessDenied)
 }
 
-/// Read a file.
-fn read(path: &Path) -> FileResult<Vec<u8>> {
+/// Reads a file from a `FileId`.
+/// It knows that whether a file is on disk or "fake".
+enum FileReader {
```

Comment:
This type seems unnecessary. Just a function `fn read(id: FileId, project_root: &Path) -> FileResult<Vec<u8>>` would suffice, wouldn't it?

Reply:
Yes, I used type to guide myself during development. It's indeed unnecessary retrospectively.